src/ui/pages/auth/Auth.tsx

```diff
@@ -12,66 +12,95 @@
+
+import styles from "./Auth.module.css";
 
 const Auth: React.FC<{}> = () => {
     const navigate = useNavigate();
 
-    // This is to avoid double prompts in dev mode because of useEffect
-    let didPrompt = false;
-
-    const getApiKeyFromUser = (): string => {
-        const apiKey = window.prompt("Please enter API key");
-
-        if (apiKey === null || apiKey.trim().length === 0) {
-            return getApiKeyFromUser();
-        }
-
-        return `${apiKey}`;
-    }
-
-    const promptAndValidateKey = async () => {
-        if (didPrompt) {
-            return;
-        }
-
-        didPrompt = true;
+    const [apiKey, setApiKey] = useState("");
+    const [apiKeyFieldError, setApiKeyFieldError] = useState("");
+    const [loading, setIsLoading] = useState<boolean>(false);
 
-        const apikeyFromUser = getApiKeyFromUser();
-        const response = await fetchDataAndRedirectIf401({
+    const validateKey = async () => {
+        setIsLoading(true);
+        const response = await fetchData({
             url: getApiUrl("/api/key/validate"),
             method: "POST",
             config: {
                 headers: {
-                    authorization: `Bearer ${apikeyFromUser}`,
+                    authorization: `Bearer ${apiKey}`,
                 },
             },
         });
 
         const body = await response.json();
 
         if (response.status === 200 && body.status === "OK") {
-            localStorageHandler.setItem(StorageKeys.API_KEY, apikeyFromUser);
+            localStorageHandler.setItem(StorageKeys.API_KEY, apiKey);
             navigate("/");
+        } else if (response.status === UNAUTHORISED_STATUS) {
+            setApiKeyFieldError("Please enter a valid API Key.")
         } else {
-            didPrompt = false;
-            promptAndValidateKey();
+            setApiKeyFieldError("An error occured while validating API Key.");
         }
+
+        setIsLoading(false);
     }
 
     useEffect(() => {
         // We delete from storage first because the user could have been redirected to auth
         // because of a 401
         localStorageHandler.removeItem(StorageKeys.API_KEY);
-        promptAndValidateKey();
     }, [])
 
+    const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+
+        if (apiKey !== null && apiKey !== undefined && apiKey.length > 0) {
+            validateKey();
+        } else {
+            setApiKeyFieldError("API Key field cannot be empty.");
+        }
+    }
+
+    const handleApiKeyFieldChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { value } = e.target;
+        setApiKey(value);
+        setApiKeyFieldError("");
+    }
+
     return (
-        <div>Loading...</div>
+        <div className={styles.pageContainer}>
+            <div className={styles.apiKeyFormContainer}>
+                <h1 className={styles.apiKeyFormTitle}>Validate your API Key</h1>
```

Comment:
"Enter your API key"

Reply:
Also remove the divider under the label, looks strange

Reply:
Updated